java/src/jmri/util/NonNullArrayList.java

```diff
@@ -1,41 +1,44 @@
 package jmri.util;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Objects;
+
 import javax.annotation.Nonnull;
 
 /**
  * An ArrayList that SpotBugs understands will never contain null elements.
  *
  * @see java.util.ArrayList
  * @see java.util.List
- * @author	Bob Jacobsen, Copyright (C) 2017
+ * @author Bob Jacobsen, Copyright (C) 2017
+ * @param <E> the class of object in the List
  */
 public class NonNullArrayList<E> extends ArrayList<E> {
 
     @Override
-    public boolean add(@Nonnull E e) {
-        if (e == null) throw new IllegalArgumentException("NonNullArrayList.addAll cannot add null item");
+    public boolean add(E e) {
```

Comment:
Could we leave the @Nonnull on the parameter?

Reply:
I removed the @Nonnull because the JDT compiler (and subsequently the Eclipse and Visual Studio IDEs) warn about the illegal redefinition of parameters when we add a @Nonnull annotation to a parameter in the implementation of a method where the superclass allows a null parameter.